core/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -42,6 +42,10 @@
         new Setting<>("cluster.routing.allocation.disk.watermark.high", "90%",
             (s) -> validWatermarkSetting(s, "cluster.routing.allocation.disk.watermark.high"),
             Setting.Property.Dynamic, Setting.Property.NodeScope);
+    public static final Setting<String> CLUSTER_ROUTING_ALLOCATION_DISK_FLOOD_STAGE_SETTING =
+        new Setting<>("cluster.routing.allocation.disk.floodstage", "95%",
+            (s) -> validWatermarkSetting(s, "cluster.routing.allocation.disk.floodstage"),
```

Comment:
I am happy to do this but it should be a sep. PR it's also tricky to do since we don't have a fully picture of the new settings when they are updated. essentially I don't think we can easily protect from this today. to make it work I think we need an extra validation round when settings are updated